lib/buffer.js

```diff
@@ -535,10 +535,35 @@ Buffer.prototype.inspect = function inspect() {
 };
 
 
-Buffer.prototype.compare = function compare(b) {
+Buffer.prototype.compare = function compare(b, start, end, thisStart, thisEnd) {
   if (!(b instanceof Buffer))
     throw new TypeError('Argument must be a Buffer');
 
+  if (typeof start !== 'undefined' &&
+      (typeof start !== 'number' ||
+       +start !== (start >> 0)))
+    throw new TypeError('"targetStart" argument must be an integer');
+  if (typeof thisEnd !== 'undefined' &&
+      (typeof thisEnd !== 'number' ||
+       +thisEnd !== (thisEnd >> 0)))
+    throw new TypeError('"sourceEnd" argument must be an integer');
+  if (typeof end !== 'undefined' &&
+      (typeof end !== 'number' ||
+       +end !== (end >> 0)))
+    throw new TypeError('"targetEnd" argument must be an integer');
+  if (typeof thisStart !== 'undefined' &&
+      (typeof thisStart !== 'number' ||
+       +thisStart !== (thisStart >> 0)))
+    throw new TypeError('"sourceStart" argument must be an integer');
+
+  if (typeof start !== 'undefined') {
+    thisStart >>= 0;
+    start >>= 0;
+    thisEnd = (isNaN(thisEnd) ? this.byteLength : thisEnd >> 0);
```

Comment:
undefined is enough. Everything else counts as a passed argument and coerced if applicable.